packages/concerto-ui-react/src/lib/concertoForm.js

```diff
@@ -162,8 +163,9 @@ class ConcertoForm extends Component {
 
   onFieldValueChange(e, key) {
     const value = e.type === 'checkbox' ? e.checked : (e.value || e.target.value);
-    this.setState(set({ ...this.state.value}, key, value));
-    this.props.onValueChange(this.state.value);
+    const valueClone = set({ ...this.state.value}, key, value);
+    this.setState(valueClone);
```

Comment:
Just double-checking -- should this not be `this.setState({ value: valueClone })` ?

Reply:
I miss my demo app! Thanks Diana, whatever would I do without you!

I've updated the story to make testing easier and I've also resolved along with some other issues that I found while testing.